### PR TITLE
Patch01

### PR DESCRIPTION
**Pull request by ggggg**
added /home

### CODE REVIEW

File: dist/settings.json [hunk 1]
```diff
@@ -3,9 +3,7 @@
     "Version": "3.0.0",
     "AnnounceInterval": 120,
     "KeepAllItemsOnDeath": false,
-    "KeptItemsOnDeath": [],
-    "LocalChatOverHead": true,
-    "LocalChatInChat": false
+    "KeptItemsOnDeath": []
```
Comment (PLASMAchicken) [inline dist/settings.json]: Why was his removed?

File: dist/settings.json [hunk 2]
```diff
@@ -24,6 +22,12 @@
         "ess"
       ]
     },
+  {
+      "CommandName": "Home",
+      "Commands": [
+        "home"
+      ]
+    },
```
Comment (PLASMAchicken) [inline dist/settings.json]: I would actually prefer if it was added at the bottom of the settings.json file as it makes writing a Change log much easier

File: dist/localization.json [hunk 3]
```diff
@@ -95,7 +95,9 @@
     "command_failed_jail": "This Command cannot be used when in jail!",
     "command_failed_crimes": "This Command cannot be used when wanted!",
     "bypass_enabled": "Bypass has been enabled.",
-    "bypass_disabled": "Bypass has been disabled."
+    "bypass_disabled": "Bypass has been disabled.",
+    "You_only_own": "You only own {0} apartment",
+    "No_Appartments": "You don't own any appartments"
```
Comment (PLASMAchicken) [inline dist/localization.json]: Use snake_case for this, also add a "." behind your sentences.

File: dist/localization.json [hunk 4]
```diff
@@ -171,6 +173,7 @@
     "player_vehicle_spawned": "Spawnado {0}.",
     "player_give": "Dado {0} {2} {1}.",
     "discord": "Nosso Discord: {0}"
+
```
Comment (PLASMAchicken) [inline dist/localization.json]: Please remove this newline if it doesn't have any use

File: src/BPEssentials/Commands/Home/Home.cs [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+using UnityEngine;
+
+namespace BPEssentials.Commands
+{
+    public class Home : Command
+    {
+        public void Invoke(ShPlayer player, int homeNumber)
+        {
+
+            List<ShApartment> apartments = new List<ShApartment>(player.ownedApartments.Keys);
```
Comment (PLASMAchicken) [inline src/BPEssentials/Commands/Home/Home.cs]: Is this really needed?

File: src/BPEssentials/Commands/Home/Home.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+                player.TS("You_only_own", apartments.Count.ToString());
+                return;
+            }
+
+            //make the apartments start at 1 (eg. /home 1 would point to apartments[0])
+            homeNumber--;
+            System.Console.WriteLine(apartments[homeNumber].GetRotation.x);
+            System.Console.WriteLine(apartments[homeNumber].GetRotation.y);
+            System.Console.WriteLine(apartments[homeNumber].GetRotation.z);
+            System.Console.WriteLine(apartments[homeNumber].GetRotation.w);
```
Comment (PLASMAchicken) [inline src/BPEssentials/Commands/Home/Home.cs]: Remove the Logs used for Debugging

File: src/BPEssentials/Commands/Home/Home.cs [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+            if (apartments[homeNumber].GetRotation.y <0.9)
+            {
+                System.Console.WriteLine("nice");
```
Comment (PLASMAchicken) [inline src/BPEssentials/Commands/Home/Home.cs]: ?

File: src/BPEssentials/Commands/Home/Home.cs [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+using BrokeProtocol.Entities;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace BPEssentials.Commands
+{
+    public class Home : Command
+    {
+        public void Invoke(ShPlayer player, int homeNumber)
```
Comment (PLASMAchicken) [inline src/BPEssentials/Commands/Home/Home.cs]:
```suggestion
        public void Invoke(ShPlayer player, int homeNumber = 0)
```
Make it default to 0.

File: dist/settings.json (hunk 1 above)
Comment (ggggg) [reply dist/settings.json]: I dont even know

Comment (ggggg) [reply dist/settings.json]: I dont think I removed it, I think it was a problem with conflicted local and git versions

File: src/BPEssentials/Commands/Home/Home.cs (hunk 8 above)
Comment (ggggg) [reply src/BPEssentials/Commands/Home/Home.cs]: Oh I will make it default to 1 (I made it that homes start at 1 and not 0

File: src/BPEssentials/Commands/Home/Home.cs (hunk 5 above)
Comment (ggggg) [reply src/BPEssentials/Commands/Home/Home.cs]: Ik its stupid, but no other way works lol (maybe arrays will work)

File: src/BPEssentials/Commands/Home/Home.cs (hunk 7 above)
Comment (ggggg) [reply src/BPEssentials/Commands/Home/Home.cs]: Debug

File: src/BPEssentials/Commands/Home/Home.cs (hunk 6 above)
Comment (ggggg) [reply src/BPEssentials/Commands/Home/Home.cs]: K

File: dist/localization.json (hunk 4 above)
Comment (ggggg) [reply dist/localization.json]: Done

File: dist/localization.json (hunk 3 above)
Comment (ggggg) [reply dist/localization.json]: Done

File: dist/settings.json (hunk 1 above)
Comment (ggggg) [reply dist/settings.json]: I added it back

File: dist/localization.json (hunk 4 above)
Comment (PLASMAchicken) [reply dist/localization.json]: No?

File: src/BPEssentials/Commands/Home/Home.cs [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+            }
+            else if (apartments.Count < homeNumber || homeNumber < 1)
+            {
+
+                player.TS("You_only_own", apartments.Count.ToString());
+                return;
+            }
+
+            //make the apartments start at 1 (eg. /home 1 would point to apartments[0])
+            homeNumber--;
```
Comment (UserR00T) [inline src/BPEssentials/Commands/Home/Home.cs]: Remove this line. See the suggestion below for more information.
```suggestion
```

File: src/BPEssentials/Commands/Home/Home.cs [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+        public void Invoke(ShPlayer player, int homeNumber=1)
+        {
+
+            List<ShApartment> apartments = new List<ShApartment>(player.ownedApartments.Keys);
+            if (apartments.Count == 0)
+            {
+                player.TS("No_Appartments");
+                return;
+            }
+            else if (apartments.Count < homeNumber || homeNumber < 1)
```
Comment (UserR00T) [inline src/BPEssentials/Commands/Home/Home.cs]: To make this more readable, switch the two sides around.
```suggestion
            if (homeNumber > apartments.Count)
```

File: src/BPEssentials/Commands/Home/Home.cs [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+            {
+
+                player.TS("You_only_own", apartments.Count.ToString());
+                return;
+            }
+
+            //make the apartments start at 1 (eg. /home 1 would point to apartments[0])
```
Comment (UserR00T) [inline src/BPEssentials/Commands/Home/Home.cs]: Remove or move the comment to the correct place after review.
```suggestion
```

File: src/BPEssentials/BPEssentials.csproj [hunk 12]
```diff
@@ -272,13 +272,17 @@
     <Reference Include="UnityEngine.XRModule">
       <HintPath>lib\Managed\UnityEngine.XRModule.dll</HintPath>
     </Reference>
+    <Reference Include="zGameSource">
+      <HintPath>lib\zGameSource.dll</HintPath>
+    </Reference>
```
[the remark this answers is not in the document]
Comment (ggggg) [reply src/BPEssentials/BPEssentials.csproj]: Its needed, I got an error if I dont and this is the fix (suggested by plasma) that works

File: src/BPEssentials/Commands/Home/Home.cs (hunk 9 above)
[the remark this answers is not in the document]
Comment (ggggg) [reply src/BPEssentials/Commands/Home/Home.cs]: No, I dont think you understand why its there, read the comment (it is in the right place), I dont want people to have to type /home 0, I rather they type /home 1 for the first home.

File: src/BPEssentials/BPEssentials.csproj (hunk 12 above)
[the remark this answers is not in the document]
Comment (PLASMAchicken) [reply src/BPEssentials/BPEssentials.csproj]: Yeah, my mistake this should have already been added in with #48. Due to the new GameSource Events Overwrite.

File: src/BPEssentials/Commands/Home/Home.cs (hunk 9 above)
[the remark this answers is not in the document]
Comment (UserR00T) [reply src/BPEssentials/Commands/Home/Home.cs]: Which is exactly what this does.

File: src/BPEssentials/Commands/Home/Home.cs [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+            }
+
+            if (apartments.Count == 0)
+            {
+                player.TS("no_appartments");
+                return;
+            }
+
+            // Make the apartments start at 1 (eg. /home 1 would point to apartments[0]).
+            homeNumber--;
```
Comment (UserR00T) [inline src/BPEssentials/Commands/Home/Home.cs]: Remove these lines and add a pre-decrement below.
```suggestion
```

File: src/BPEssentials/Commands/Home/Home.cs [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+            var offset = new Vector3(-1, 0, -1);
+            var apartment = apartments[Math.Max(0, homeNumber)];
```
Comment (UserR00T) [inline src/BPEssentials/Commands/Home/Home.cs]: Add a pre-decrement.
```suggestion
            var apartment = apartments[Math.Max(0, --homeNumber)];
```

File: src/BPEssentials/Commands/Home/Home.cs [hunk 15]
```diff
@@ -29,7 +29,7 @@ public void Invoke(ShPlayer player, int homeNumber = 1)
             // Make the apartments start at 1 (eg. /home 1 would point to apartments[0]).
             homeNumber--;
```
Comment (UserR00T) [inline src/BPEssentials/Commands/Home/Home.cs]: Remove these lines, as you're already doing this action below now.
```suggestion
```

**Pull request merged**